pkg/base1/cockpit.js

```diff
@@ -2154,44 +2154,37 @@ function basic_scope(cockpit, jquery) {
             cockpit.info.dispatchEvent("changed");
     };
 
-    function User() {
-        var self = this;
-        event_mixin(self, { });
-
-        self["user"] = null;
-        self["name"] = null;
-
-        var dbus = cockpit.dbus(null, { "bus": "internal" });
-        dbus.call("/user", "org.freedesktop.DBus.Properties",
-                  "GetAll", [ "cockpit.User" ],
-                  { "type": "s" })
-            .done(function(reply) {
-                var user = reply[0];
-                self["user"] = user.Name.v;
-                self["name"] = user.Full.v;
-                self["id"] = user.Id.v;
-                self["groups"] = user.Groups.v;
-                self["home"] = user.Home.v;
-                self["shell"] = user.Shell.v;
-            })
-            .fail(function(ex) {
-                console.warn("couldn't load user info: " + ex.message);
-            })
-            .always(function() {
-                dbus.close();
-                self.dispatchEvent("changed");
-            });
-    }
-
     var the_user = null;
-    Object.defineProperty(cockpit, "user", {
-        enumerable: true,
-        get: function user_get() {
-            if (!the_user)
-                the_user = new User();
-            return the_user;
+    cockpit.user = function () {
+        var dfd = cockpit.defer();
+        if (!the_user) {
+            var dbus = cockpit.dbus(null, { "bus": "internal" });
```

Comment:
Due to javascript hoisting insanity, we typically try not to define variables inside an if block, just moving the var above works.